### DISCUSSION TITLE
Eth lib agnostic web3modal - idea

### DISCUSSION THREAD

**Discussion by glitch-txs** [WalletConnect/walletconnect-monorepo#2637] [Ideas]
Hi Team! :) As you may know I've been helping devs (mainly frontend) to integrate walletconnect in their dapps and gathered some information/feedback for an idea I think it could be useful :)

There are currently two great packages for frontend integration:

- web3modal with wagmi
- ethereum-provider

This packages are great! But there's a third option that seems needed and not currently filled. Web3modal with wagmi shows browser wallets on the modal but is opinionated in terms of ethereum libraries, since it uses viem by default. Ethereum-provider on the other hand is unopinionated in this matter, but it doesn't show browser wallets on the modal. So there are some devs that fall in between: They want the modal with browser wallets but maybe they want to use ethers.js or web3.js instead, could be because of personal preference or because the base code they are working on is written with these libraries.

It would be cool to have this options combined into a library where devs can see browser wallets on the modal and use any ethereum library of choice. 

Hope this is helpful! 👍

**Comment by xzilja** [comment]
@glitch-txs watch out for V3 😉

**Reply by glitch-txs** [reply]
🔥💯

**Reply by jxom** [reply]
👀

**Reply by glitch-txs** [reply]
Hi @0xAsimetriq I was thinking also adding up to this about a feature for the modal, mainly thinking about [this problem with Safe](https://github.com/orgs/WalletConnect/discussions/2733), I think it could be cool if in the same modal the user could be able to choose what chains want to connect to, so the in code side, the config on the chains could be left empty, also I'd consider moving the chain setup from the init fn to the the connect function instead for customization. Also, thinking on the idea of selecting the chains for the session on the same modal, there could be in the code-side a default chain param in config to fallback if the user doesn't select any, not sure if makes sense, just throwing some idea n.n
